guidance/library/_json.py

```diff
@@ -435,6 +440,10 @@ def json(
             - A JSON schema object. This is a JSON schema string which has been passed to ``json.loads()``
             - A subclass of ``pydantic.BaseModel``
             - An instance of ``pydantic.TypeAdapter``
+
+    compact : bool
```

Comment:
I'm glad you wrote that notebook. It is definitely evidence that validates my own subjective experience.

The new lexed grammar makes it really easy to leave the whitespace up to the LLM, but it doesn't come without costs... There are many fewer forced tokens when we don't force a specific whitespace convention, and on a preliminary benchmark I ran on my laptop, generating a JSON "fighter" takes twice as long with flexible whitespace vs without...

(In a later PR) it could be nice to support forcing whitespace that conforms to `json.dump(..., indent=n)` for various values of n?

@riedgar-ms quick question -- do you think compact should be True or False by default? And is the kwarg named appropriately?

Reply:
I think `compact` is as good a name as any. If the speed difference is that large, it might be better to have the default be `True` and note in the docs that if generation quality is poor, people might want to experiment with `compact=False`

Reply:
On this note: 

> json.dump(..., indent=n)

having _any_ fixed whitespace format should be approximately identical in speed to the compact one, since all the whitespace tokens get forced anyway. I think when we get around to changing defaults, we should pick a reasonably well-represented pretty printing format (probably identical to indent=2 or indent=4 in python json.dump terms) which has empirically reasonable perplexity on a variety of tasks. I don't think the compact representation right now should really be defaulted/advertised for anything longer term, because generating "pretty printed" style JSON is essentially 0 additional cost for a fixed format.

Running @nopdive's langsmith benchmarks here on a few models, once this is parametrizable, is probably enough to give us a reasonable heuristic for what the default whitespace number should look like

Reply:
@Harsha-Nori I think I'd like to implement specific whitespace formats as a separate PR (I think it's going to be just nontrivial enough to warrant this). 

That being said, `compact` mode is quite a bit faster than whitespace-flexible mode (although both are considerably faster than current guidance's compact mode). I'm inclined to leave it in, even if just so we can have something to benchmark with (knowing that any specific whitespace format should have similar performance). Are you ok with leaving `compact` in, maybe `False` by default?